src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -553,6 +556,7 @@ protected override void OnClear()
                 }
 
                 labelPlaces.Clear();
+                nodesSelected = new Dictionary<string, string>();
```

Comment:
No, it is only called when the `Workspace` is cleared or if there is a change to `DynamoModel`'s property. Modifications to the nodes do not call `OnClear()`.